docs/learn/bossroom/architecture.md

```diff
@@ -0,0 +1,266 @@
+
+After those experiments, we established new rules for the team:
+
+* Use domain-based assemblies
+* Use single classes for small components (for example, the Boss Room door with a simple on/off state).
+* If a class never grows too big, use a single NetworkBehaviour (because it’s easy to maintain).
+* Use client and server classes (with each pointing to the other) for client/server separation.
+* Place client/server pairs in the same assembly.
+* If you start the game as a client, the server components disable themselves, leaving you with only client components executing. Ensure you don’t destroy the server components. NGO still requires the server component for network message sending.
+* The clients have a `m_Server`, and servers have a `m_Client` property.
+* The `Server` class owns server-driven `NetworkVariables`. Similarly, the `Client` class owns owner-driven `NetworkVariables`. This ownership separation helps make larger classes more readable and maintainable.
+* Use partial classes when separating ownership isn’t possible.
+* Continue using the `Client`/`Server` prefix to make contexts more obvious. Note: You can’t use prefixes for `ScriptableObjects` that have file name requirements.
+* Use `Client`/`Server`/`Shared` separation when you have a one-to-many relationship where your server class needs to send information to many client classes. You can also achieve this with the [`NetworkedMessageChannel`](#heading=h.wfrn29pfpphk).
+
+You still need to take care of code executing in `Start` and `Awake`. If the code in `Start` and `Awake` runs simultaneously with the NetworkManager's initialization, it might not know whether the player is a host or client.
+
+### Publisher-Subscriber Messaging
+
+We implemented a dependency injection-friendly Publisher-Subscriber pattern that allows Boss Room to send and receive strongly-typed messages in a loosely-coupled manner, where communicating systems only know about the `IPublisher`/`ISubscriber` of a given message type.
+
+Because publishers and subscribers are classes, we can have more interesting behavior for message transfer, such as buffered and networked messaging. See the NetworkedMessageChannel section.
+
+These mechanisms allow Boss Room to avoid circular references and limit the dependency surface between assemblies. Cross-communicating systems rely on common messages but don't necessarily need to know about each other. Because messages don’t need to know about each other, we can more easily separate them into smaller assemblies.
+
+These mechanisms allow for strong separation of concerns and coupling reduction using PubSub and dependency injection (DI). We use DI to pass the handles to the `IPublisher` or `ISubscriber` of any given event type. As a result, message publishers and consumers are unaware of each other.
+
+`MessageChannel` classes implement the `IPublisher` and `ISubscriber` interfaces and provide the actual messaging logic.
+
+#### `NetworkedMessageChannel`
```

Comment:
```suggestion
`MessageChannel` classes implement the `IPublisher` and `ISubscriber` interfaces and provide the actual messaging logic.

:::note
Boss Room could have used a 3rd party library for messaging (like MessagePipe). Since we needed custom networked channels and our use cases were quite simple, we decided to go with our own light implementation. If we had had a full game with more gameplay, we might have switched to a library that had more features.
:::

#### `NetworkedMessageChannel`
```